x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/RestDeleteExpiredDataAction.java

```diff
@@ -41,9 +44,34 @@ public String getName() {
 
     @Override
     protected RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
-        DeleteExpiredDataAction.Request request = restRequest.hasContent() ?
-            DeleteExpiredDataAction.Request.PARSER.apply(restRequest.contentParser(), null) :
-            new DeleteExpiredDataAction.Request();
+        DeleteExpiredDataAction.Request request;
+        if (restRequest.hasContent()) {
+            request = DeleteExpiredDataAction.Request.PARSER.apply(restRequest.contentParser(), null);
+        } else {
+            request = new DeleteExpiredDataAction.Request();
+
+            String perSecondParam = restRequest.param(DeleteExpiredDataAction.Request.REQUESTS_PER_SECOND.getPreferredName());
+            if (perSecondParam != null) {
+                try {
+                    request.setRequestsPerSecond(Float.parseFloat(perSecondParam));
+                } catch (NumberFormatException e) {
+                    throw new IllegalArgumentException("Failed to parse float parameter [" +
+                        DeleteExpiredDataAction.Request.REQUESTS_PER_SECOND.getPreferredName() +
+                        "] with value [" + perSecondParam + "]", e);
+                }
+            }
+
+            String timeoutParam = restRequest.param(DeleteExpiredDataAction.Request.TIMEOUT.getPreferredName());
+            if (timeoutParam != null) {
+                request.setTimeout(restRequest.paramAsTime(timeoutParam, null));
+            }
+        }
+
+        String jobId = restRequest.param(Fields.JOB_ID.getPreferredName());
```

Comment:
This should pull in the field name from the `Job` config object. Not the data frame analytics fields.